src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -221,6 +222,128 @@ def cli_cosmosdb_list(client, resource_group_name=None):
     return client.list()
 
 
+def cli_cosmosdb_network_rule_list(client, resource_group_name, account_name):
+    """ Lists the virtual network accounts associated with a Cosmos DB account """
+    cosmos_db_account = client.get(resource_group_name, account_name)
+    return cosmos_db_account.virtual_network_rules
+
+
+def _get_virtual_network_id(cmd, resource_group_name, subnet, virtual_network):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    from msrestazure.tools import is_valid_resource_id, resource_id
+    if not is_valid_resource_id(subnet):
+        if virtual_network is None:
+            raise CLIError("Provide either the subnet ID or the VNET and subnet name")
+        subnet = resource_id(
+            subscription=get_subscription_id(cmd.cli_ctx),
+            resource_group=resource_group_name,
+            namespace='Microsoft.Network', type='virtualNetworks',
+            name=virtual_network, child_type_1='subnets', child_name_1=subnet
+        )
+    return subnet
+
+
+def cli_cosmosdb_network_rule_add(cmd,
+                                  client,
+                                  resource_group_name,
+                                  account_name,
+                                  subnet,
+                                  virtual_network=None,
+                                  ignore_missing_vnet_service_endpoint=False):
+    """ Adds a virtual network rule to an existing Cosmos DB database account """
+    subnet = _get_virtual_network_id(cmd, resource_group_name, subnet, virtual_network)
+    existing = client.get(resource_group_name, account_name)
+
+    virtual_network_rules = []
+    rule_already_exists = False
+    for rule in existing.virtual_network_rules:
+        if rule.id != subnet:
+            virtual_network_rules.append(
+                VirtualNetworkRule(id=rule.id,
+                                   ignore_missing_vnet_service_endpoint=rule.ignore_missing_vnet_service_endpoint))
+        else:
+            if rule.ignore_missing_vnet_service_endpoint == ignore_missing_vnet_service_endpoint:
+                raise CLIError("This rule already exists for the Cosmos DB account")
```

Comment:
My recommendation to simplify the logic here would be to just detect that the rule already exists and then to add a warning "This rule already exists and will be overwritten." instead of then checking whether the flag changed or not.

Generally, we would like creates to be idempotent, so throwing an error because nothing has changed would make this command not idempotent and could interfere with scripting efforts.